File: src/com/pokosho/bot/FileStudyBot.java
package com.pokosho.bot;

import java.io.BufferedReader;
import java.io.FileReader;

import com.pokosho.PokoshoException;

/**
 * ファイルパスから文字列を読み込み学習する基本的なBot.
 * @author kaiba
 *
 */
public class FileStudyBot extends AbstractBot {

	public FileStudyBot(String dbPropPath, String botPropPath) throws PokoshoException {
		super(dbPropPath, botPropPath);
	}

	@Override
	public void study(String file) throws PokoshoException {
		try {
			FileReader freader = new FileReader(file);
			BufferedReader reader = new BufferedReader(freader);
			String line = null;
			do {
				line = reader.readLine();
				studyFromLine(line);
			} while (line != null);
		} catch (Exception e) {
			throw new PokoshoException(e);
		}
	}
}

File: src/main/kotlin/com/pokosho/bot/FileStudyBot.kt
package com.pokosho.bot

import com.pokosho.PokoshoException
import java.io.BufferedReader
import java.io.FileReader

/**
 * ファイルパスから文字列を読み込み学習する基本的なBot.
 * @author kaiba
 */
class FileStudyBot @Throws(PokoshoException::class)
constructor(botPropPath: String) : AbstractBot(botPropPath) {

    @Throws(PokoshoException::class)
    override fun study(str: String?) {
        try {
            FileReader(str).use { fileReader ->
                BufferedReader(fileReader).use { reader ->
                    var message: String?
                    do {
                        message = reader.readLine()
                        studyFromMessage(message)
                    } while (message != null)
                }
            }
        } catch (e: Exception) {
            throw PokoshoException(e)
        }

    }
}
